mantle/platform/util.go

```diff
@@ -34,14 +34,16 @@ import (
 // session on the Machine m. Manhole blocks until the shell session has ended.
 // If os.Stdin does not refer to a TTY, Manhole returns immediately with a nil
 // error.
-func Manhole(m Machine) error {
+func Manhole(m Machine) (err error) {
 	fd := int(os.Stdin.Fd())
 	if !terminal.IsTerminal(fd) {
 		return nil
 	}
 
 	tstate, _ := terminal.MakeRaw(fd)
-	defer terminal.Restore(fd, tstate)
+	defer func() {
+		err = terminal.Restore(fd, tstate)
```

Comment:
To expound on @lucab's review....this is almost, but not quite. 

As written, this will swallow the error. In the event that `err != nil` _and_ the terminal fails to restore, then the returned error will be that of `terminal.Restore`.

I think a better solution would be:
```
defer func() {
      e := terminal.Restore(fd, state)
      if err != nil {
          return fmt.Errorf("%v; %v", err, e)
      }
      return e
}
```

Reply:
Make sense. Thank you for review.